src/controllers/api/classController.js

```diff
@@ -7,36 +7,13 @@ const Spell = require('../../models/spell');
 const Feature = require('../../models/feature');
 const Proficiency = require('../../models/proficiency');
 const utility = require('./utility');
+const SimpleController = require('../simpleController');
 
-exports.index = async (req, res, next) => {
-  const search_queries = {};
-  if (req.query.name !== undefined) {
-    search_queries.name = { $regex: new RegExp(utility.escapeRegExp(req.query.name), 'i') };
-  }
+const simpleController = new SimpleController(Class);
 
-  await Class.find(search_queries)
-    .sort({ index: 'asc' })
-    .then(data => {
-      res.status(200).json(utility.NamedAPIResource(data));
-    })
-    .catch(err => {
-      next(err);
-    });
-};
+exports.index = async (req, res, next) => await simpleController.index(req, res, next);
```

Comment:
I think this could be simplified to:

`exports.index = simpleController.index`

Reply:
Very true, missed this one as I was going through the refactor.